### PR TITLE
Preferences: Support setting any dashboard as home, not just the starred ones

### PR DESCRIPTION
**Pull request by ryantxu**
This PR is really motivated by trying to clean up nested ID use rather than UIDs.  The preferences API currently sends both ID and UID.

This PR:
* replaces the frontend code to work directly with UIDs
* removes the custom select that limited us to only our starred dashboards
* updates the backend APIs so that sending an empty string is the same as sending 0

### CODE REVIEW

File: public/app/core/components/SharedPreferences/SharedPreferences.test.tsx [hunk 1]
```diff
@@ -193,7 +192,7 @@ describe('SharedPreferences', () => {
       timezone: 'browser',
       weekStart: '',
       theme: '',
-      homeDashboardUID: undefined,
```
Comment (ivanortegaalba) [inline public/app/core/components/SharedPreferences/SharedPreferences.test.tsx:196]: We're removing the `Default` option from the select, which allows the user to set the default Grafana home page. Is this intentional?

|Default value selected|Home page when Default is selected|
|-|-|
Captura de Pantalla 2022-08-26 a las 10 01 14|Captura de Pantalla 2022-08-26 a las 10 01 05

File: public/app/core/components/SharedPreferences/SharedPreferences.tsx [hunk 2]
```diff
@@ -210,14 +171,11 @@ export class SharedPreferences extends PureComponent<Props, State> {
...
               >
-                <Select
-                  value={dashboards.find((dashboard) => dashboard.uid === homeDashboardUID)}
-                  getOptionValue={(i) => i.uid}
-                  getOptionLabel={this.getFullDashName}
-                  onChange={(dashboard: SelectableValue<DashboardSearchItem>) =>
-                    this.onHomeDashboardChanged(dashboard.uid)
-                  }
-                  options={dashboards}
+                <DashboardPicker
```
Comment (ivanortegaalba) [inline public/app/core/components/SharedPreferences/SharedPreferences.tsx:174]: We probably can remove the hint. It doesn't make sense anymore

Captura de Pantalla 2022-08-26 a las 10 04 13

**Review by dprokop — APPROVED** [review]
Looks good and works as expected. Agree with @ivanortegaalba's comments 👍

File: public/app/core/components/SharedPreferences/SharedPreferences.tsx (hunk 2 above)
Comment (ryantxu) [reply public/app/core/components/SharedPreferences/SharedPreferences.tsx:160]: done

File: public/app/core/components/SharedPreferences/SharedPreferences.test.tsx (hunk 1 above)
Comment (ryantxu) [reply public/app/core/components/SharedPreferences/SharedPreferences.test.tsx:196]: We now use the standard "x" to clear the custom home dashboard and go back to the default one.  I updated the language so it is now:

**Review by sakjur — APPROVED** [review]
Backend changes LGTM

**Review by ivanortegaalba — APPROVED** [review]
All working perfectly well! 💯

**Pull request merged**